src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -633,7 +634,8 @@ protected function prepareException(Throwable $e)
     {
         return match (true) {
             $e instanceof BackedEnumCaseNotFoundException => new NotFoundHttpException($e->getMessage(), $e),
-            $e instanceof ModelNotFoundException => $e->isNotFound() ? new NotFoundHttpException($e->getMessage(), $e) : new HttpException($e->getStatus(), $e->getMessage(), $e),
+            $e instanceof ModelNotFoundException => new NotFoundHttpException($e->getMessage(), $e),
+            $e instanceof InvalidIdFormatException => new HttpException(422, $e->getMessage(), $e),
```

Comment:
You could use `\Symfony\Component\HttpFoundation\Request::HTTP_UNPROCESSABLE_ENTITY` here
```suggestion
        return $this->status === Request::HTTP_UNPROCESSABLE_ENTITY;
```